var/spack/repos/builtin/packages/vtk-m/package.py

```diff
@@ -124,6 +124,7 @@ class VtkM(CMakePackage, CudaPackage, ROCmPackage):
     conflicts("+rocm", when="@:1.6")
     conflicts("+rocm", when="+cuda")
     conflicts("+rocm", when="~kokkos", msg="VTK-m does not support HIP without Kokkos")
+    conflicts("+rocm", when="+virtuals", msg="VTK-m does not support virtual functions with ROCm")
```

Comment:
Virtual methods are obsolete and have been completely removed from VTK-m 2.0. I suggest we remove this option altogether.

Reply:
I think that can be expressed in the variant itself and described in the variant description, then everything else can just be left as is.